modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3548,14 +3548,24 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
             Tuple containing:
                 1) 2-d NumPy array of aligned left partitions
                 2) list of 2-d NumPy arrays of aligned right partitions
-                3) joined index along ``axis``
-                4) List with sizes of partitions along axis that partitioning
-                   was done on. This list will be empty if and only if all
+                3) joined index along ``axis``, may be ``ModinIndex`` if not materialized
+                4) If materialized, list with sizes of partitions along axis that partitioning
+                   was done on, otherwise ``None``. This list will be empty if and only if all
                    the frames are empty.
         """
         if isinstance(other, type(self)):
             other = [other]
 
+        if not force_repartition and all(
+            o._check_if_axes_identical(self, axis) for o in other
```

Comment:
`o._check_if_axes_identical(self, axis) for o in other` -> `self._check_if_axes_identical(o, axis) for o in other`? I am just wondering why the logic is inverted a bit?

Reply:
there's no reason for this, both of the options would perform identically